### PR TITLE
Correction of the definition of word token in the grok parser.

### PR DESCRIPTION
**Pull request by KateYoak**
With the help of Tess Neau:
- the regex rule behind the word token is indeed \b\w+\b (https://github.com/DataDog/logs-backend/blob/prod/domains/event-platform/libs/processing/processing-parsing/src/main/resources/grok_patterns/core#L6)

### Merge instructions

- [ ] Please merge after reviewing

### CODE REVIEW

File: content/en/logs/log_configuration/parsing.md [hunk 1]
```diff
@@ -106,7 +106,7 @@ Here is a list of all the matchers and filters natively implemented by Datadog:
 : Matches an integer number (with scientific notation support) and parses it as an integer number.
 
 `word`
-: Matches characters from a-z, A-Z, 0-9, including the _ (underscore) character.
+: Matches "a word", which starts with a word boundary, contains characters from a-z, A-Z, 0-9, including the _ (underscore) character and ends with a word boundary. Equivalent to `\b\w*\b` in regex.
```
Comment (brett0000FF) [inline content/en/logs/log_configuration/parsing.md:109]:
```suggestion
: Matches a _word_, which starts with a word boundary; contains characters from a-z, A-Z, 0-9, including the `_` (underscore) character; and ends with a word boundary. Equivalent to `\b\w*\b` in regex.
```
Small edit for clarity.

**Review by brett0000FF — APPROVED** [review]

**Comment by brett0000FF** [comment]
@KateYoak - Thanks for this update! Let me know if this is ready to merge. 👍

File: content/en/logs/log_configuration/parsing.md [hunk 2]
```diff
@@ -106,7 +106,7 @@ Here is a list of all the matchers and filters natively implemented by Datadog:
 : Matches an integer number (with scientific notation support) and parses it as an integer number.
 
 `word`
-: Matches characters from a-z, A-Z, 0-9, including the _ (underscore) character.
+: Matches a _word_, which starts with a word boundary; contains characters from a-z, A-Z, 0-9, including the `_` (underscore) character; and ends with a word boundary. Equivalent to `\b\w*\b` in regex.
```
Comment (KateYoak) [inline content/en/logs/log_configuration/parsing.md:109]:
```suggestion
: Matches a _word_, which starts with a word boundary; contains characters from a-z, A-Z, 0-9, including the `_` (underscore) character; and ends with a word boundary. Equivalent to `\b\w+\b` in regex.
```

**Review by KateYoak — COMMENTED** [review]
Just realized that the regex in the code is \b\w+\b, not \b\w*\b. Corrected.

**Comment by KateYoak** [comment]
Thanks @brett0000FF for a fast review!  I committed your suggestion and also made one more commit, correcting the regex I initially put in.

**Comment by KateYoak** [comment]
It should be good to merge.

**Pull request merged**